server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1927,7 +1928,8 @@ DiscoveryNode getNode() {
      * Else it configures the size to 80% of available capacity for a dedicated search node, if not explicitly defined.
      */
     private void initializeFileCache(Settings settings, CircuitBreaker circuitBreaker) throws IOException {
-        if (DiscoveryNode.isSearchNode(settings)) {
+        boolean isWritableRemoteIndexEnabled = FeatureFlags.isEnabled(FeatureFlags.WRITEABLE_REMOTE_INDEX_SETTING);
+        if (DiscoveryNode.isSearchNode(settings) || isWritableRemoteIndexEnabled) {
             NodeEnvironment.NodePath fileCacheNodePath = nodeEnvironment.fileCacheNodePath();
             long capacity = NODE_SEARCH_CACHE_SIZE_SETTING.get(settings).getBytes();
```

Comment:
We'll need to revisit the cache initialization logic here. we'll to move to a dynamic criteria for capacity eventually.